README.md

```diff
@@ -152,6 +152,7 @@ oid = repo.write("This is a blob.", :blob)
 index = repo.index
 index.read_tree(repo.head.target.tree)
 index.add(:path => "README.md", :oid => oid, :mode => 0100644)
+index.write
```

Comment:
This use-case does not want to persist the index, it's using it as a throw-away data structure (though it'd probably be good to replace `index = repo.index` to `index = Index.new`.